colorizer_data/types.py

```diff
@@ -97,15 +100,64 @@ class FrameDimensions(TypedDict):
     """Height of a frame in physical units (not pixels)."""
 
 
-class DatasetMetadata(TypedDict):
+class BaseMetadataJson(TypedDict):
+    """JSON dictionary format for `BaseMetadata`."""
+
+    name: str
+    description: str
+    author: str
+    dateCreated: str
+    lastModified: str
+    revision: str
+    dataVersion: str
+
+
+@dataclass
+class BaseMetadata:
+    """Shared metadata between datasets and collection files."""
+
+    name: str = None
+    description: str = None
+    author: str = None
+    date_created: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
+    last_modified: str = None
+    """Formatted datetime string. See `DATETIME_FORMAT`. """
+    revision: int = None
+    """
+    Revision number. Will be updated each time the dataset or collection
+    is rewritten. Starts at 0.
+    """
+    data_version: str = CURRENT_VERSION
+    """Version of the data writer utility scripts. Uses semantic versioning (e.g. v1.0.0)"""
+
+    def to_json(self) -> BaseMetadataJson:
```

Comment:
there has got to be a more automatic way to do this - it's just transforming the property names